projects/bp-gallery/src/components/views/picture/face-tagging/FaceTag.tsx

```diff
@@ -0,0 +1,143 @@
+import { Cancel, OpenWith } from '@mui/icons-material';
+import { CSSProperties, useContext, useMemo } from 'react';
+import '../../../../shared/style.scss';
+import { AuthRole, useAuth } from '../../../provider/AuthProvider';
+import { PictureViewContext } from '../PictureView';
+import { useFaceTagging } from '../../../../hooks/context-hooks';
+import { useImageRect } from './helpers/image-rect';
+
+const triangleHeight = 10;
+const triangleWidth = 20;
+const boundary = 0.9;
+
+export type FaceTagData = {
+  id: string | undefined;
+  personTagId: string | undefined | null;
+  name: string;
+  x: number;
+  y: number;
+  noPointerEvents?: boolean;
+};
+
+export const FaceTag = ({
+  data: { id, personTagId, name, x, y, noPointerEvents },
+}: {
+  data: FaceTagData;
+}) => {
+  const { role } = useAuth();
+  const context = useFaceTagging();
+  const isFaceTagging = context?.isFaceTagging;
+  const handleDelete = async () => {
+    if (id === undefined) {
+      return;
+    }
+    await context?.removeTag(id);
+  };
+
+  const handleMove = async () => {
+    if (personTagId === undefined) {
+      return;
+    }
+    await handleDelete();
+    context?.setActiveTagId(personTagId);
+  };
```

Comment:
Gibt es einen Grund wieso ihr direkt den Tag löscht, würde es nicht reichen einfach x und y zu updaten? Die Relation zur Person und Bild bleiben ja gleich

Reply:
das war einfach die leichteste Methode um handleMove zu implementieren, da wir nur bereits implementierte Methoden nutzen konnten, anstatt das nochmal implementieren zu müssen.

Reply:
Ist im großen ganzen nicht allzu wichtig, aber kann mir vorstellen dass dadurch die id's relativ schnell nach oben steigen und wir da große Lücken zwischen wirklich  existierenden Tags haben werden  und Strapi muss jedes mal 3 tables anstatt einen anfassen

Reply:
Ich glaube nicht, dass die Kuratoren so häufig FaceTags verschieben. Wenn man wirklich das existierende tag weiterverwenden wollen würde, müsste man immer unterscheiden, ob man gerade einen existierenden oder neuen tag in der hand hält. Außerdem gibt's dann den edge case, dass man noch ein existierendes tag in der Hand hat, aber den Modus verlässt. Heißt das dann löschen oder dahin zurück wo er war? Ich wäre dafür, das so zu lassen, wie es ist.